src/main.ts

```diff
@@ -35,17 +45,58 @@ export class Minifier {
     }
   }
 
+  setTypeChecker(typeChecker: ts.TypeChecker) { this.typeChecker = typeChecker; }
+
+  reportError(n: ts.Node, message: string) {
+    var file = n.getSourceFile();
+    var fileName = file.fileName;
+    var start = n.getStart(file);
+    var pos = file.getLineAndCharacterOfPosition(start);
+    var fullMessage = `${fileName}:${pos.line + 1}:${pos.character + 1}: ${message}`;
+    this.errors.push(fullMessage);
+    if (minifierOptions.failFast) {
+      throw new Error(fullMessage);
+    }
+  }
+
   // Recursively visits every child node, emitting text of the sourcefile that is not a part of
   // a child node.
   visit(node: ts.Node) {
     switch (node.kind) {
       case ts.SyntaxKind.PropertyAccessExpression: {
+        let pae = <ts.PropertyAccessExpression>node;
+        let exprSymbol = this.getExpressionSymbol(pae);
+        let output = '';
+        let children = pae.getChildren();
+
+        output += this.visit(pae.expression);
+        output += pae.dotToken.getText();
+
+        // Early exit when exprSymbol is undefined.
+        if (!exprSymbol) {
+          this.reportError(pae.name, 'Symbol information could not be extracted.\n');
+          return;
+        }
+
+        var isExternal = exprSymbol.declarations.some(
+            (decl) => !!(decl.getSourceFile().fileName.match(/\.d\.ts/)));
+        if (isExternal) {
```

Comment:
nit: prefer to keep if statements that fit on one line on one line (for single statement, no braces, no else ifs).